kroxylicious-filters/kroxylicious-record-encryption/src/main/java/io/kroxylicious/filter/encryption/dek/DekManager.java

```diff
@@ -66,7 +67,14 @@ public CompletionStage<Dek<E>> generateDek(@NonNull K kekRef, @NonNull CipherMan
         Objects.requireNonNull(kekRef);
         Objects.requireNonNull(cipherManager);
         return kms.generateDekPair(kekRef)
-                .thenApply(dekPair -> new Dek<>(dekPair.edek(), DestroyableRawSecretKey.toDestroyableKey(dekPair.dek()), cipherManager, maxEncryptionsPerDek));
+                .thenApply(dekPair -> {
+                    DestroyableRawSecretKey destroyableKey = DestroyableRawSecretKey.toDestroyableKey(dekPair.dek());
+                    if (destroyableKey.numKeyBits() < cipherManager.requiredNumKeyBits()) {
+                        throw new EncryptionConfigurationException("KMS returned " + destroyableKey.numKeyBits() + "-bit DEK but "
+                                + cipherManager.name() + " requires keys of " + cipherManager.requiredNumKeyBits() + " bits");
```

Comment:
I think this test is a good sanity check but how does this failure mode propagate to users?  

I think this is going to trigger a flood of requests. A request to the KMS for every message % parallelism  produced as we ask the KMS for a new key which will likely get the same result until a human intervenes. I worry this is a good way to blow up a could service bill and a bad way to get human attention. 

That said I don't have any good ways of addressing that in the short term so maybe this should just be an issue in the backlog?

Reply:
DekManager could maybe stand down the kekRef on the assumption it isn't going to change in a hurry

Reply:
I imagine in future we will make KMS aware of the desired keySize so it could apply it's own resilience logic. But the DekManager should protect itself from misbehaving KMS anyway.

Reply:
We need to tighten up the contract of `Kms`, either explicitly by making the key requirements part of the signature, or alternatively just stating in the javadoc that we require 256 bit keys. 

We have other requirements as well as key size. For example, I think we're also assuming (somewhere) that the cipher supports the `RAW` key format.

I'll open an issue for this.

Reply:
https://github.com/kroxylicious/kroxylicious/issues/1425 FTR

Reply:
I think #1425 is a good idea but that wasn't really what I was getting at (or at least I see them as independent). Do we want to have the DekManager handle the failing key generation in this PR or should we just raise an issue to manage it later?

Reply:
separate issue I think, the same thing would happen in the existing 0.6.0 code right? We'd hit a cipher init problem with a 128bit key and the `AES_256/..` transformation.

Reply:
Have opened #1426 for hardening DekManager against a misbehaving KMS